### PR TITLE
Use clang-format-14 for formatting

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

File: src/details/ArborX_Callbacks.hpp [hunk 1]
```diff
@@ -30,7 +30,9 @@ enum class CallbackTreeTraversalControl
 namespace Details
 {
 
-struct [[deprecated]] InlineCallbackTag{};
+struct [[deprecated]] InlineCallbackTag
+{
+};
```
Comment (dalg24) [inline src/details/ArborX_Callbacks.hpp:35]: Yuk

File: examples/molecular_dynamics/example_molecular_dynamics.cpp [hunk 2]
```diff
@@ -94,7 +94,7 @@ int main(int argc, char *argv[])
         particles[4 * id + 3] = {i * dx + .5f, j * dy + .5f, k * dz + .0f};
       });
 
-  Kokkos::View<float * [3], MemorySpace> velocities(
+  Kokkos::View<float *[3], MemorySpace> velocities(
```
Comment (dalg24) [inline examples/molecular_dynamics/example_molecular_dynamics.cpp:97]: I like that new formatting of view data type with mixed dynamic and static dimensions

File: .clang-format [hunk 3]
```diff
@@ -1,7 +1,26 @@
...
+  AfterUnion: true
+  AfterExternBlock: true
+  BeforeCatch: true
+  BeforeElse: true
+  BeforeLambdaBody: false
+  BeforeWhile: true
+  IndentBraces: false
+  SplitEmptyFunction: true
+  SplitEmptyRecord: true
+  SplitEmptyNamespace: true
```
Comment (aprokop) [inline .clang-format:23]: Do we need to provide all of the values, or just the changed ones?

Comment (masterleinad) [reply .clang-format]: You need all of them. We can't just say "use this style; but change those values". I don't know what the default for `Custom` is, so I would provide everything.

File: src/ArborX_BruteForce.hpp [hunk 4]
```diff
@@ -55,7 +55,7 @@ class BruteForce
             typename CallbackOrView, typename View, typename... Args>
   std::enable_if_t<Kokkos::is_view<std::decay_t<View>>{}>
   query(ExecutionSpace const &space, Predicates const &predicates,
-        CallbackOrView &&callback_or_view, View &&view, Args &&... args) const
+        CallbackOrView &&callback_or_view, View &&view, Args &&...args) const
```
[the remark this answers is not in the document]
Comment (masterleinad) [reply src/ArborX_BruteForce.hpp:58]: https://stackoverflow.com/questions/65964384/how-to-make-clang-format-11-put-a-space-after-variadic-template-decleration

**Comment by masterleinad** [comment]
I'm reasonably happy with this now.

**Comment by aprokop** [comment]
> I'm reasonably happy with this now.

Mostly OK from my perspective. I would probably prefer the old variant 
```c++
Kokkos::parallel_for("blah",
  blah
);
```
to
```c++
Kokkos::parallel_for(
  "blah",
  blah
);

**Comment by aprokop** [comment]
@dalg24 @Rombur you ok with the current state?

**Comment by Rombur** [comment]
That's fine. I only care that `template <typename T>` is on a different line than the function and than `{` and `}` are aligned on the same column in most cases. Everything else is fine with me.

**Review by dalg24 — APPROVED** [review]

**Pull request merged**